src/main/java/com/iemr/hwc/controller/wo/LocationControllerWo.java

```diff
@@ -154,4 +157,40 @@ public String getLocDetailsBasedOnSpIDAndPsmIDNew(@RequestBody String comingRequ
         }
         return response.toString();
     }
+
+    @CrossOrigin
+    @ApiOperation(value = "Updating village coordinates collected by nurse", consumes = "application/json", produces = "application/json")
+    @RequestMapping(value = { "/update/geolocation/wo" }, method = { RequestMethod.POST })
+    public String updateGeolocationVillage(@RequestBody String requestObj) {
+        OutputResponse response = new OutputResponse();
+        try {
+            logger.info("Request object for Geolocation update :" + requestObj);
+
+            JsonObject jsnOBJ = new JsonObject();
+            JsonParser jsnParser = new JsonParser();
+            JsonElement jsnElmnt = jsnParser.parse(requestObj);
+            jsnOBJ = jsnElmnt.getAsJsonObject();
+
+            JsonElement jElmtLatitude = jsnOBJ.get("latitude");
+            JsonElement jElmtLongitude = jsnOBJ.get("longitude");
+            JsonElement jElmtDistrictBranchID = jsnOBJ.get("districtBranchID");
+
+
+            if (jsnOBJ != null && jElmtLatitude!=null && jElmtLongitude!=null && jElmtDistrictBranchID!=null) {
+                int responseUpdate = locationServiceImpl.updateGeolocationByDistrictBranchID(jElmtLatitude.getAsDouble(), jElmtLongitude.getAsDouble(), jElmtDistrictBranchID.getAsInt());
+                if(responseUpdate==1){
+                    response.setResponse(responseUpdate+"");
+                }
+                else if(responseUpdate==101){
+                    response.setError(5000, "Unable to update data");
```

Comment:
change message to "Unable to update the lat long for an active record"

Reply:
Done